src/core/ddsi/src/ddsi_nwpart.c

```diff
@@ -0,0 +1,375 @@
+        break;
+      case DNAR_DISTANT:
+      case DNAR_UNREACHABLE:
+        nwpart_iter_error (it, tok, "address does not match a local interface");
+        break;
+    }
+  }
+
+  if (!nwpart_iter_ok (it))
+    return;
+  else if ((**nextpp = ddsrt_malloc (sizeof (***nextpp))) == NULL)
+    nwpart_iter_error (it, tok, "out of memory");
+  else
+  {
+    (**nextpp)->loc = loc_to_use;
+    (**nextpp)->loc.port = port;
+    (**nextpp)->next = NULL;
+    DDSRT_WARNING_MSVC_OFF(6011);
+    *nextpp = &(**nextpp)->next;
+    DDSRT_WARNING_MSVC_ON(6011);
+  }
+}
+
+static bool nwpart_iter_fini (struct nwpart_iter *it)
+{
+  if (it->msgtag)
+    ddsrt_free (it->msgtag);
+  return it->ok;
+}
+
+static void convert_network_partition_addresses_one (struct nwpart_iter *npit, struct ddsi_config_networkpartition_listelem *np, uint32_t port_mc, uint32_t port_data_uc, const char *tok)
+{
+  // FIXME: it'd be nice if the one could specify a port and additional sockets would be created
+  ddsi_locator_t loc;
+  switch (ddsi_locator_from_string (npit->gv, &loc, tok, npit->gv->m_factory))
+  {
+    case AFSR_OK:       break;
+    case AFSR_INVALID:  nwpart_iter_error (npit, tok, "not a valid address"); return;
+    case AFSR_UNKNOWN:  nwpart_iter_error (npit, tok, "unknown address"); return;
+    case AFSR_MISMATCH: nwpart_iter_error (npit, tok, "address family mismatch"); return;
+  }
+  if (loc.port != 0)
+    nwpart_iter_error (npit, tok, "no port number expected");
+  else if (ddsi_is_mcaddr (npit->gv, &loc))
+    nwpart_iter_append_address (npit, tok, &loc, port_mc);
+  else if (strspn (np->interface_names, ", \t") != strlen (np->interface_names))
+    nwpart_iter_error (npit, tok, "unicast addresses not allowed when interfaces are also specified");
+  else
+    nwpart_iter_append_address (npit, tok, &loc, port_data_uc);
+}
+
+static int convert_network_partition_addresses (struct ddsi_domaingv *gv, uint32_t port_data_uc)
+{
+  const uint32_t port_mc = ddsi_get_port (&gv->config, DDSI_PORT_MULTI_DATA, 0);
+  struct nwpart_iter npit;
+  nwpart_iter_init (&npit, gv, ": partition address");
+  struct ddsi_config_networkpartition_listelem *np;
+  while ((np = nwpart_iter_next (&npit)) != NULL)
+  {
+    char *copy = ddsrt_strdup (np->address_string), *cursor = copy, *tok;
+    while ((tok = ddsrt_strsep (&cursor, ",")) != NULL)
+      convert_network_partition_addresses_one (&npit, np, port_mc, port_data_uc, tok);
+    ddsrt_free (copy);
+  }
+  return nwpart_iter_fini (&npit) ? 0 : -1;
+}
+
+static int convert_network_partition_interfaces (struct ddsi_domaingv *gv, uint32_t port_data_uc)
+{
+  struct nwpart_iter npit;
+  nwpart_iter_init (&npit, gv, ": interface name");
+  struct ddsi_config_networkpartition_listelem *np;
+  while ((np = nwpart_iter_next (&npit)) != NULL)
+  {
+    char *copy = ddsrt_strdup (np->interface_names), *cursor = copy, *tok;
+    while ((tok = ddsrt_strsep (&cursor, ",")) != NULL)
```

Comment:
Is whitespace already stripped out of `np->interface_names`? If not, `strcmp` on line 339 might fail because of this?

Reply:
I figured it would be acceptable to treat all white space in the names as significant and that there really was no reason to allow empty entries. I'm open to changing it, it just seems like a needless complication in the code.

Reply:
I was triggered by the call to `strspn` in `convert_network_partition_addresses_one` that uses `", \t"`, but accepting leading/trailing white space in a name sounds reasonable, so I'm fine with the current code.